Editor/CoreAPI/StateManager.cs

```diff
@@ -1,13 +1,14 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
```

Comment:
I know a lot of this has changed since the comments on #95, so I'm going to hold off reviewing this until #95 gets merged in and this PR is rebased.